src/test/java/GmailServletTest.java

```diff
@@ -0,0 +1,119 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import com.google.api.services.gmail.model.Message;
+import com.google.sps.model.AuthenticationVerifier;
+import com.google.sps.model.GmailClient;
+import com.google.sps.model.GmailClientFactory;
+import com.google.sps.servlets.GmailServlet;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+
+/**
+ * Test Gmail Servlet to ensure response contains correctly parsed messageIds. Assumes
+ * AuthenticatedHttpServlet is functioning properly (those tests will fail otherwise).
+ */
+@RunWith(JUnit4.class)
+public final class GmailServletTest {
+  private static final AuthenticationVerifier authenticationVerifier =
+      Mockito.mock(AuthenticationVerifier.class);
+  private static final GmailClientFactory gmailClientFactory =
+      Mockito.mock(GmailClientFactory.class);
+  private static final GmailClient gmailClient = Mockito.mock(GmailClient.class);
+  private static final GmailServlet servlet =
+      new GmailServlet(authenticationVerifier, gmailClientFactory);
+
+  private static final boolean AUTHENTICATION_VERIFIED = true;
+  private static final String ID_TOKEN_KEY = "idToken";
+  private static final String ID_TOKEN_VALUE = "sampleId";
+  private static final String ACCESS_TOKEN_KEY = "accessToken";
+  private static final String ACCESS_TOKEN_VALUE = "sampleAccessToken";
+  private static final Cookie sampleIdTokenCookie = new Cookie(ID_TOKEN_KEY, ID_TOKEN_VALUE);
+  private static final Cookie sampleAccessTokenCookie =
+      new Cookie(ACCESS_TOKEN_KEY, ACCESS_TOKEN_VALUE);
+  private static final Cookie[] validCookies =
+      new Cookie[] {sampleIdTokenCookie, sampleAccessTokenCookie};
+
+  private HttpServletRequest request;
+  private HttpServletResponse response;
+  private StringWriter stringWriter;
+  private PrintWriter printWriter;
+
+  private static final String MESSAGE_ID_ONE = "messageIdOne";
+  private static final String MESSAGE_ID_TWO = "messageIdTwo";
+  private static final String MESSAGE_ID_THREE = "messageIdThree";
+  private static final List<Message> noMessages = new ArrayList<>();
+  private static final List<Message> threeMessages =
+      Arrays.asList(
+          new Message().setId(MESSAGE_ID_ONE),
```

Comment:
FYI: If `Message` is immutable and you use `ImmutableList.of`, this will effectively become a constant

Reply:
Awesome! Thanks